sky/tools/create_macos_framework.py

```diff
@@ -152,6 +157,30 @@ def process_framework(dst, args, fat_framework, fat_framework_binary):
     ],
                           cwd=dst)
 
+  macos_filepath_with_entitlements = ''
```

Comment:
Should lines 160 to 170 be moved to line 151 and remove the zipping subprocess? as it is we are zipping twice.

Reply:
Sure, updated to zip only once. Yeah I remember this artifact was historically unzipped twice (https://github.com/christopherfujino/codesign.py/blob/master/codesign.py#L151) for some reasons, but verified just now and it looks there we can do a single zip/unzip.

Also it's interesting that if i double click to extract the artifact, the symlinks are not preserved. But if i use the subprocess command "unzip.. -d .." then the symlinks are preserved in the extracted file.